src/Core/Injector/Injector.php

```diff
@@ -587,7 +587,7 @@ protected function instantiate($spec, $id = null, $type = null)
         if ((!$type || $type !== self::PROTOTYPE)
             && empty($constructorParams)
             && is_subclass_of($class, DataObject::class)) {
-            $constructorParams = array(null, true);
+            $constructorParams = array(null, DataObject::CREATE_SINGLETON);
```

Comment:
I have a feeling this should really be a part of a DataObjectFactory. We'll have other places where similar concerns require logic in a similar fashion. This ties injector to ORM more than it should IMO.

Maybe allowing us to specify a factory for a class hierarchy would be good to look into?

Reply:
```yaml
SilverStripe\Core\Injector\Injector:
  SilverStripe\ORM\DataObject:
    factory: SilverStripe\ORM\DataObjectFactory
    inherits: true
```

php: 

```php
public function getFactory($spec) {
    // Get factory for this class
    if (isset($spec['factory'])) {
        return $this->get($spec['factory']);
    }
    // Look for any factory on parent class with `inherits: true`
    $next = $spec['class'];
    while ($next = get_parent_class($next)) {
        $nextConfig = $this->configLocator->locateConfigFor($next);
        if (isset($nextConfig['factory']) && !empty($nextConfig['inherits'])) {
            return $this->get($nextConfig['factory']));
        }
     }
    // Default constructor
    return $this->getObjectCreator();
}
```

Reply:
I don't necessarily disagree with any of this, but it's unrelated to this PR. This PR just removed the use of a now-deprecated value in the constructor.

Perhaps you want to pick this up as a separate issue.

Briefly, the reasons I haven't dug into this are:
 * Concerns about adding complexity for one weird edge-case
 * Performance
 * Whether the "best fix" would be an API breakage and as such best left 'til SS5

None of those are insurmountable, but they've meant that I've left this one alone myself.